attiny402_firmware/src/main.ino

```diff
@@ -0,0 +1,16 @@
+#include "serial.h"
+
+void setup() {
+    Serial.begin(115200);
+    pinMode(A3, OUTPUT);
+    digitalWrite(A3, LOW);
+    pinMode(A2, INPUT);
+}
+
+void loop() {
+    handler ser;
```

Comment:
I _think_ that putting the constructor here will have the nice effect of calling it **every time loop is called**. Basically:
```
loop{
    //call constructor
    //use ser
    //destroy ser
    //repeat
}
```
though I'm not sure because of that `while(1)` (which btw I'd replace with `while(true)`)

Reply:
why would I need to call and destroy the constructor at each loop?

Reply:
You don't, that's what _I think_ it's currently happening (at least if you get out of the while loop)

Reply:
Marking this as solved since you changed the code